src/android/LocalNotification.java

```diff
@@ -132,57 +158,228 @@ public boolean execute (final String action, final JSONArray args,
+                } else if (action.equals("clear")) {
                     clear(args, command);
-                } else
-                if (action.equals("clearAll")) {
+                } else if (action.equals("clearAll")) {
                     clearAll(command);
-                } else
-                if (action.equals("type")) {
+                } else if (action.equals("type")) {
                     type(args, command);
-                } else
-                if (action.equals("ids")) {
+                } else if (action.equals("ids")) {
                     ids(args, command);
-                } else
-                if (action.equals("notification")) {
+                } else if (action.equals("notification")) {
                     notification(args, command);
-                } else
-                if (action.equals("notifications")) {
+                } else if (action.equals("notifications")) {
                     notifications(args, command);
+                } else if (action.equals("hasDoNotDisturbPermissions")) {
+                    hasDoNotDisturbPermissions(command);
+                } else if (action.equals("requestDoNotDisturbPermissions")) {
+                    requestDoNotDisturbPermissions(command);
+                } else if (action.equals("isIgnoringBatteryOptimizations")) {
+                    isIgnoringBatteryOptimizations(command);
+                } else if (action.equals("requestIgnoreBatteryOptimizations")) {
+                    requestIgnoreBatteryOptimizations(command);
+                } else if (action.equals("dummyNotifications")) {
+                    dummyNotifications(command);
                 }
             }
         });
 
         return true;
     }
 
+    /**
+     * Required for Android 13 to get the runtime notification permissions.
+     *
+     * @param command The callback context used when calling back into JavaScript.
+     */
+    private void dummyNotifications(CallbackContext command) {
+
+        fireEvent("dummyNotifications");
+        NotificationManager mNotificationManager;
+        NotificationCompat.Builder mBuilder;
+        String NOTIFICATION_CHANNEL_ID = "10004457";
+        String notificationMsg = "Test";
+        String notificationTitle = "Mdd";
+        Context context =  cordova.getActivity().getApplicationContext();
+
+        Intent intentToLaunch = new Intent(context, TriggerReceiver.class);
+        intentToLaunch.putExtra("Callfrom", "reminders");
+
+        final PendingIntent resultPendingIntent = PendingIntent.getActivity(context,
+                0, intentToLaunch, PendingIntent.FLAG_IMMUTABLE);
+
+        mBuilder = new NotificationCompat.Builder(context,NOTIFICATION_CHANNEL_ID);
+        mBuilder.setContentIntent(resultPendingIntent);
+
+        mNotificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
+
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O){
```

Comment:
This method has only an effect on android >= 8. The code above should be moved in this if-statement, because it has no effect, if the if-statement is not executed.

Reply:
moved it inside the if-case